### PR TITLE
feat: breaking changes

### PR DESCRIPTION
**Pull request by artemrys**
BREAKING CHANGE: remove ESCU workflow and run unit tests before build

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -749,7 +766,7 @@ jobs:
           {
           echo "argo-server=argo.wfe.splgdi.com:443"
           echo "argo-http1=true"
-          echo "argo-secure=true"
+          echo "argo-secure=true"ƒ
```
Comment (mkolasinski-splunk) [inline .github/workflows/reusable-build-test-release.yml:769]: What is the last sign in this line?

File: .github/workflows/reusable-build-test-release.yml [hunk 2]
```diff
@@ -2168,25 +2213,6 @@ jobs:
           path: |
             ${{ needs.setup.outputs.directory-path }}/diag*
 
-  validate-pr-title:
```
Comment (mkolasinski-splunk) [inline .github/workflows/reusable-build-test-release.yml:2171]: Please provide reasons for removing this job.

File: .github/workflows/reusable-build-test-release.yml (hunk 1 above)
Comment (artemrys) [reply .github/workflows/reusable-build-test-release.yml]: Nice catch, fixed it!

File: .github/workflows/reusable-build-test-release.yml (hunk 2 above)
Comment (artemrys) [reply .github/workflows/reusable-build-test-release.yml:2171]: There are pros and cons for removing this job.

Pros of removing:

* It takes seconds to run, but it costs us 1 minute of billing time
* It is a miscellaneous job, totally optional. If a developer knows what they are doing, this job is not needed and I think we should expect this

Cons of removing:

* Enforces consistency for the PR title, but not for the commits

Comment (mkolasinski-splunk) [reply .github/workflows/reusable-build-test-release.yml:2171]: - I think rather than removing this job we could move it to be a step in `setup` job, so it will not consume 1 minute of billing time.

- Sure we can expect that, but this can serve for example as a reminder in case someone forgot to add proper title.

- Enforcing consistency of PR title together with "Squash and merge" for PR results in clear overview of changes history, e.g.: https://github.com/splunk/splunk-add-on-for-cisco-meraki/commits/main

To summarize, I think following this action doesn't require significant human effort or computational resources if it's moved as a step to another job. Nonetheless, it provides value, so I prefer to retain it.

Comment (artemrys) [reply .github/workflows/reusable-build-test-release.yml:2171]: Okay, I'll leave it, as a follow up let's have only "Squash and merge" option available for the add-ons. I'll take this on me.
